src/drt/src/dr/FlexGridGraph.cpp

```diff
@@ -221,26 +235,23 @@ void FlexGridGraph::initEdges(
     } else {
       nonPrefLayerNum = layerNum;
     }
-    yIdx = 0;
-    for (auto& [yCoord, ySubMap] : yMap) {
-      auto yIt = ySubMap.find(layerNum);
-      auto yIt2 = ySubMap.find(layerNum + 2);
-      auto yIt3 = ySubMap.find(nonPrefLayerNum);
-      bool yFound = (yIt != ySubMap.end());
-      bool yFound2 = (yIt2 != ySubMap.end());
-      bool yFound3 = (yIt3 != ySubMap.end());
-      xIdx = 0;
-      for (auto& [xCoord, xSubMap] : xMap) {
-        auto xIt = xSubMap.find(layerNum);
-        auto xIt2 = xSubMap.find(layerNum + 2);
-        auto xIt3 = xSubMap.find(nonPrefLayerNum);
-        bool xFound = (xIt != xSubMap.end());
-        bool xFound2 = (xIt2 != xSubMap.end());
-        bool xFound3 = (xIt3 != xSubMap.end());
-        // add cost to out-of-die edge
-        bool isOutOfDieVia = outOfDieVia(xIdx, yIdx, zIdx, dieBox_);
-        // add edge for preferred direction
-        if (dir == dbTechLayerDir::HORIZONTAL && yFound) {
+    auto& yLayerMap = yMap[layerNum];
+    auto& yLayer2Map = yMap[layerNum + 2];
```

Comment:
What if layerNum is the top routing layer? Wouldn't that insert layerNum+2 as a new entry to the flatmap and possible reallocate memory?

Reply:
Yes, but it's cheap:
* it only happens once at most,
* it may not reallocate,
* even if it does, it's on the order of 10 elements,
* and all the inner maps are just moved so they themselves don't reallocate.

In the original code this can happen for each coordinate.